validator/submain.py

```diff
@@ -69,6 +69,9 @@ def timeout_handler(signum, frame):
             package = open(path, 'rb')
             test_package(err, package, path, expectation, for_appversions)
 
+        if err.detected_type == PACKAGE_EXTENSION:
+            err.metadata['is_extension'] = True
```

Comment:
The linter bit is a good reasoning. I don't think there's any other code-path, `submain` is pretty common to be called, imho it's always called but I could easily be wrong.